src/main/java/com/maria/library/facade/mapper/BookMapper.java

```diff
@@ -15,4 +18,8 @@ public Book mapToEntity(BookDto bookDto) {
         book.setPublicationYear(bookDto.getPublicationYear());
         return book;
     }
+    public BookDetailsDto mapToEntity(Book book, Author author, Category category) {
+        BookDetailsDto bdo = new BookDetailsDto(book.getTitle(), book.getPublicationYear(), category.getName(), author.getFirstname() + " " + author.getLastname());
```

Comment:
also, there is no need for 2 lines of code, refactor this method to have just one line containing return statement